presto-tablestore/src/main/java/com/facebook/presto/tablestore/TablestoreRecordCursor4RangeBlock.java

```diff
@@ -0,0 +1,288 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tablestore;
+
+import com.alicloud.openservices.tablestore.model.ColumnType;
+import com.alicloud.openservices.tablestore.model.SimpleRowMatrixBlockRowIterator;
+import com.facebook.presto.common.type.CharType;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.common.type.VarbinaryType;
+import com.facebook.presto.common.type.VarcharType;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.RecordCursor;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+import org.apache.commons.lang3.tuple.Triple;
+
+import javax.annotation.Nonnull;
+
+import java.util.List;
+
+public class TablestoreRecordCursor4RangeBlock
+        extends TablestoreAbstractRecordCursor
+        implements RecordCursor
+{
+    private final SimpleRowMatrixBlockRowIterator rowIterator;
+
+    public TablestoreRecordCursor4RangeBlock(TablestoreFacade tablestoreFacade, ConnectorSession session, TablestoreSplit split,
+            List<TablestoreColumnHandle> columnHandles)
+    {
+        super(tablestoreFacade, session, split, columnHandles);
+        Triple<SimpleRowMatrixBlockRowIterator, Long, Long> x = tablestoreFacade.rowIterator4RangeBlock(session, split, columnHandles);
+        this.rowIterator = x.getLeft();
+        apiNano += x.getRight();
+    }
+
+    @Override
+    public long getCompletedBytes()
+    {
+        checkClosed();
+        return rowIterator.getCompletedBytes();
+    }
+
+    @Override
+    protected boolean hasNext()
+    {
+        return rowIterator.hasNext();
+    }
+
+    @Override
+    protected void next()
+    {
+        rowIterator.next(); //有可能报错
```

Comment:
remove the comment.  we can add a try/catch if you think the exception should be handled here.

Reply:
done